testsuite/gateways/apicast/selfmanaged.py

```diff
@@ -47,12 +49,23 @@ def __new__(cls, staging: bool, openshift: OpenShiftClient, settings,
                     if subclass.__name__ == kind:
                         kind = subclass  # type: ignore
                         break
+
+            def _openshift(subclass):
+                """get right openshift for a subclass"""
+                custom = weakget(settings)[subclass.__name__]["openshift"] % None
+                if not custom:
+                    return openshift
+                if isinstance(custom, OpenShiftClient):
+                    return custom
+                return OpenShiftClient(
```

Comment:
I don't understand why this is needed

Reply:
Default openshift is passed to `SelfManagerApicast` that is not openshift that for example `OperatorApicast` should have as it can have (and it really does have) custom openshift settings. Direct glitch is that `OpenshiftClient.project_exists` returns wrong value (3scale namespace vs. apicast-operator namespace).

Reply:
Yeah, but why do you create OpenShiftClient directly, if configured properly it should always be an OpenShiftClient already.

Reply:
I don't think so, if `SelManagedApicast` is used, `OperatorApicast` section shouldn't be touched and processed. This part is happening here

Reply:
Contrary I can't imagine openshift instance could be ever here, anyway one `if` doesn't hurt (at least here, at least for now)